lib/datadog/tracing/contrib/propagation/sql_comment/comment.rb

```diff
@@ -0,0 +1,37 @@
+# typed: false
+require 'erb'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Propagation
+        module SqlComment
+          class Comment
+            def initialize(hash)
+              @hash = hash
+            end
+
+            def to_s
+              @string ||= begin
+                ret = String.new
+
+                @hash.each do |key, value|
+                  next if value.nil?
+
+                  value = ERB::Util.url_encode(value) # url encode
```

Comment:
I recommend using the built-in [CGI.escape](https://rubyapi.org/2.7/o/cgi/util#method-i-escape) method.